sphinxcontrib/needs/directives/needfilter.py

```diff
@@ -58,10 +60,10 @@ def run(self):
             'docname': env.docname,
             'lineno': self.lineno,
             'target_node': targetnode,
-            'show_tags': self.options.get("show_tags", False) is None,
-            'show_status': self.options.get("show_status", False) is None,
-            'show_filters': self.options.get("show_filters", False) is None,
-            'show_legend': self.options.get("show_legend", False) is None,
+            'show_tags': self.options.get("show_tags", False),
```

Comment:
I think this is not the same logic.

The option is None, if the key exists.
Goal of the replace code was to get True or False, now it will be set to None.

I think ``show_tags in self.options.keys()`` is what you want, if a replacement is needed.

Reply:
But maybe Sphinx as changed it's logic for boolean values in the meantime.
Not sure as the code got created during Sphinx 1.8.

Reply:
ah i see! will update.

one way to capture this would be to make the 'options' dictionary a class, instead of a plain `dict`. It would be a little verbose, because you don't yet have access to [dataclasses](https://docs.python.org/3/library/dataclasses.html) (requires Python 3.7+). Another option would be to use [Pydantic](https://pydantic-docs.helpmanual.io/), but that still requires Python 3.6+.
The advantage of this approach is if you tried to assign `None` to a field typed as `bool` (which is what i'm erroneously doing here) than your IDE and linter would complain

Anywhere that you changed a dictionary to a class, and that dictionary is exposed in the API, that would be a breaking change. so probably not something to do lightly.